src/ini.zig

```diff
@@ -0,0 +1,1076 @@
+                        @intCast(line_offset + @as(i32, @intCast(eq_sign_idx)) + 1),
+                        .value,
+                    );
+                    break :brk Expr.init(E.String, E.String{ .data = "" }, Loc.Empty);
+                }
+                break :brk Expr.init(E.Boolean, E.Boolean{ .value = true }, Loc.Empty);
+            };
+
+            const value: Expr = switch (value_raw.data) {
+                .e_string => |s| if (bun.strings.eqlComptime(s.data, "true"))
+                    Expr.init(E.Boolean, E.Boolean{ .value = true }, Loc.Empty)
+                else if (bun.strings.eqlComptime(s.data, "false"))
+                    Expr.init(E.Boolean, E.Boolean{ .value = false }, Loc.Empty)
+                else if (bun.strings.eqlComptime(s.data, "null"))
+                    Expr.init(E.Null, E.Null{}, Loc.Empty)
+                else
+                    value_raw,
+                else => value_raw,
+            };
+
+            if (is_array) {
+                if (head.get(key)) |val| {
+                    if (val.data != .e_array) {
+                        var arr = E.Array{};
+                        arr.push(this.arena.allocator(), val) catch bun.outOfMemory();
+                        head.put(this.arena.allocator(), key, Expr.init(E.Array, arr, Loc.Empty)) catch bun.outOfMemory();
+                    }
+                } else {
+                    head.put(this.arena.allocator(), key, Expr.init(E.Array, E.Array{}, Loc.Empty)) catch bun.outOfMemory();
+                }
+            }
+
+            // safeguard against resetting a previously defined
+            // array by accidentally forgetting the brackets
+            var was_already_array = false;
+            if (head.get(key)) |val| {
+                if (val.data == .e_array) {
+                    was_already_array = true;
+                    val.data.e_array.push(this.arena.allocator(), value) catch bun.outOfMemory();
+                    head.put(this.arena.allocator(), key, val) catch bun.outOfMemory();
+                }
+            }
+            if (!was_already_array) {
+                head.put(this.arena.allocator(), key, value) catch bun.outOfMemory();
+            }
+        }
+    }
+
+    fn prepareStr(
+        this: *Parser,
+        ropealloc: Allocator,
+        val_: []const u8,
+        offset_: i32,
+        comptime usage: enum { section, key, value },
+    ) !switch (usage) {
+        .value => Expr,
+        .section => *Rope,
+        .key => []const u8,
+    } {
+        var offset = offset_;
+        var val = std.mem.trim(u8, val_, " \n\r\t");
+
+        if (isQuoted(val)) out: {
+            // remove single quotes before calling JSON.parse
+            if (val.len > 0 and val[0] == '\'') {
+                val = if (val.len > 1) val[1 .. val.len - 1] else val[1..];
+                offset += 1;
+            }
+            const src = bun.logger.Source.initPathString("<unknown>", val);
```

Comment:
we could keep the original filename if there's invalid JSON embedded within right? imagine if you're seeing an error here and the error is in `<unknown>`